lib/cloudregister/registerutils.py

```diff
@@ -173,6 +175,96 @@ def clear_rmt_as_scc_proxy_flag():
         pass
 
 
+# ----------------------------------------------------------------------------
+def clean_payg_extensions():
+    """Remove non free extensions files from /etc/products.d/."""
+    extensions = get_extensions()
+    products_paths = glob.glob('/etc/products.d/*.prod')
+
+    for extension in extensions:
+        # The following not free condition checks on
+        # the access capability of the extension
+        # This is a server side capability
+        if not extension.get('free', True):
+            extension_path = os.path.join(
+                os.sep, 'etc', 'products.d', '{}.prod'.format(
+                    extension.get('identifier')
+                )
+            )
+            if extension_path in products_paths:
+                os.unlink(extension_path)
+                logging.info(
+                    'Non free extension file %s removed' % extension_path
+                )
+
+
+# ----------------------------------------------------------------------------
+def get_product_tree(product_file=BASE_PRODUCT_PATH):
+    """
+    Read product element from baseproduct and return an etree
+    """
+    if os.path.isfile(product_file):
+        with open(product_file, 'r') as product:
+            raw_xml = product.read()
+            product_entry_point = raw_xml.index('<product')
+            return etree.fromstring(raw_xml[product_entry_point:])
+
+
+# ----------------------------------------------------------------------------
+def get_product_triplet(product_tree):
+    """
+    Extract identifier, version and arch information from
+    the given product tree
+    """
+    product_type = namedtuple(
+        'product_type', ['name', 'version', 'arch']
+    )
+    return product_type(
+        name=product_tree.find('name').text,
+        version=product_tree.find('version').text,
+        arch=product_tree.find('arch').text
+    )
+
+
+# ----------------------------------------------------------------------------
+def get_extensions():
+    """Get the extensions of the base product on the update server."""
+    prod_data = get_product_data()
+    return prod_data.get('extensions', [])
+
+
+# ----------------------------------------------------------------------------
+def get_product_data(registration_target=None):
```

Comment:
This is being moved from the command implementation for no apparent reason related to the clean up but then also not used in the command implementation....?????

Reply:
as the method s/`clean_payg_extensions`/`clean_non_free_extensions` uses these methods
tests would have failed

I guess we can try to mock those calls but seemed good to have them tested as, at the time of this PR, there were no tests for `registercloudguest`

Reply:
also, `get_extensions` can get re-used in `registercloudguest`

Reply:
@rjschwei I'm about to close this thread as resolved now as I believe Jesus provided the needed information. As I said from my perspective the involved refactoring/code-moves are good for the requested change

If this change should be split into a refactor first, let's discuss at the workshop. I would be good as I think the code move plus tests is still good to review